Code/Framework/AzFramework/AzFramework/API/ApplicationAPI.h

```diff
@@ -202,6 +202,37 @@ namespace AzFramework
         virtual bool ShouldBlockLevelLoading([[maybe_unused]]const char* levelName) { return false; }
     };
     using LevelSystemLifecycleRequestBus = AZ::EBus<LevelSystemLifecycleRequests>;
+
+    class LevelSystemLifecycleNotifications
+        : public AZ::EBusTraits
+    {
+    public:
+        // Bus Configuration
+        static const AZ::EBusHandlerPolicy HandlerPolicy = AZ::EBusHandlerPolicy::Multiple;
+        static const AZ::EBusAddressPolicy AddressPolicy = AZ::EBusAddressPolicy::Single;
+
+        // Called when loading a level fails due to it not being found.
+        virtual void OnLevelNotFound([[maybe_unused]] const char* levelName) {}
+
+        // Called after ILevelSystem::PrepareNextLevel() completes.
+        virtual void OnPrepareNextLevel([[maybe_unused]] const char* levelName) {}
+
+        // Called after ILevelSystem::OnLoadingStart() completes, before the level actually starts loading.
+        virtual void OnLoadingStart([[maybe_unused]] const char* levelName) {}
```

Comment:
In terms of the `AudioSystem` one place where we do need to make an update would be in "AudioSystemGemSystemComponent.cpp".  Currently it's registering as an `ISystemEventListener` and hooking off of `ESYSTEM_EVENT_LEVEL_LOAD_START` and `ESYSTEM_EVENT_LEVEL_POST_UNLOAD` events.  I believe we can remove those and replace them with these `OnLoadingStart` and `OnUnloadComplete` notifications.

A second spot is in "ATL.cpp", where it's also registering an `ISystemEventListener` but this time there are some additional events that it's using.  We should be able to find replacements for them.  We can chat on the side about it.